lib/phoenix_html/form.ex

```diff
@@ -303,12 +305,13 @@ defmodule Phoenix.HTML.Form do
   fallback to parameters and finally fallback to the default
   struct/map value.
   """
-  def input_value(%{source: source, impl: impl} = form, field) when is_atom(field) do
+  @spec input_value(t | atom, atom | String.t) :: term
+  def input_value(%{source: source, impl: impl} = form, field) do
     try do
       impl.input_value(source, form, field)
     rescue
       UndefinedFunctionError ->
-        case Map.fetch(form.params, Atom.to_string(field)) do
+        case Map.fetch(form.params, to_string(field)) do
```

Comment:
This change unfortunately allows other kinds of inputs. We should still support only atoms or strings.

Reply:
I would still like to avoid the dynamic dispatch. Let's create a private function called `field_to_string` and make sure all places invoke it.